src/couch/src/couch_file.erl

```diff
@@ -702,18 +703,25 @@ monitored_by_pids() ->
 % System dbs aren't monitored by couch_stats_process_tracker
 is_idle(#file{is_sys=true}) ->
     case monitored_by_pids() of
-        [] -> true;
+        [] -> lock(self());
         _ -> false
     end;
 is_idle(#file{is_sys=false}) ->
     Tracker = whereis(couch_stats_process_tracker),
     case monitored_by_pids() of
-        [] -> true;
-        [Tracker] -> true;
+        [] -> lock(self());
+        [Tracker] -> lock(self());
         [_] -> exit(tracker_monitoring_failed);
         _ -> false
     end.
 
+lock(Fd) -> try
+                ets:new(list_to_atom(pid_to_list(Fd)), [named_table]), true
```

Comment:
This doesn't pass my smell test. Using ets tables as locks has got to be asking for trouble. Between the system limit on tables and no guaranteed cleanup that's a bunch of bad behaviors just waiting to happen.

Reply:
I pulled a new one using register, without ets!

Reply:
While register does solve some of those issues that's gonna take a significant amount of work to characterize the performance. I don't think it was ever intended for this sort of usage. I'd like to see some fairly extensive benchmarking before even considering accepting this and also probably some fairly detailed writeup of the register internals explaining how this doesn't become a global lock across shard lookups.